node.gyp

```diff
@@ -888,12 +888,46 @@
     {
       'target_name': 'cctest',
       'type': 'executable',
-      'dependencies': [ 'deps/gtest/gtest.gyp:gtest' ],
+
+      'dependencies': [
+        '<(node_core_target_name)',
+        'deps/gtest/gtest.gyp:gtest',
+        'deps/cares/cares.gyp:cares',
+        'deps/zlib/zlib.gyp:zlib',
+        'node_js2c#host',
+      ],
+
       'include_dirs': [
         'src',
+        'tools/msvs/genfiles',
         'deps/v8/include',
-        '<(SHARED_INTERMEDIATE_DIR)'
+        'deps/cares/include',
+        'deps/uv/include',
+        '<(SHARED_INTERMEDIATE_DIR)', # for node_natives.h
+      ],
+
+      'libraries': [
+        '<(PRODUCT_DIR)/obj.target/node/gen/node_javascript.o',
+        '<(PRODUCT_DIR)/obj.target/node/src/node_debug_options.o',
+        '<(PRODUCT_DIR)/obj.target/node/src/async-wrap.o',
+        '<(PRODUCT_DIR)/obj.target/node/src/env.o',
+        '<(PRODUCT_DIR)/obj.target/node/src/node.o',
+        '<(PRODUCT_DIR)/obj.target/node/src/node_buffer.o',
+        '<(PRODUCT_DIR)/obj.target/node/src/node_i18n.o',
+        '<(PRODUCT_DIR)/obj.target/node/src/debug-agent.o',
+        '<(PRODUCT_DIR)/obj.target/node/src/util.o',
+        '<(PRODUCT_DIR)/obj.target/node/src/string_bytes.o',
+        '<(PRODUCT_DIR)/obj.target/node/src/string_search.o',
+        '<(PRODUCT_DIR)/obj.target/node/src/stream_base.o',
+        '<(PRODUCT_DIR)/obj.target/node/src/node_dtrace.o',
+        '<(PRODUCT_DIR)/obj.target/node/src/node_constants.o',
+        '<(PRODUCT_DIR)/obj.target/node/src/node_revert.o',
+        '<(PRODUCT_DIR)/obj.target/node/src/tracing/agent.o',
+        '<(PRODUCT_DIR)/obj.target/node/src/tracing/node_trace_buffer.o',
+        '<(PRODUCT_DIR)/obj.target/node/src/tracing/node_trace_writer.o',
+        '<(PRODUCT_DIR)/obj.target/node/src/tracing/trace_event.o',
```

Comment:
I tried this out but it did not work for me as the object files could not be found using `OBJ_DIR`. I created https://github.com/nodejs/node/pull/11857, can you take a look and see what you think about it?